task8/ProfitApp/src/main/java/com/bsu/fpmi/ProfitApp/Ad.java

```diff
@@ -0,0 +1,159 @@
+package com.bsu.fpmi.ProfitApp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.List;
+
+public class Ad {
+
+    private String id;
+    private String description;
+    private String createdAt;
+    private String link;
+    private String vendor;
+    private String photoLink;
+    private List<String> hashTags;
+    private String discount;
+    private String validUntil;
+    private Double rating;
+    private List<String>reviews;
+
+    private boolean isDateCorrect(String date) {
+        SimpleDateFormat df = new SimpleDateFormat("dd.MM.yyyy");
+        df.setLenient(false);
+        try {
+            df.parse(date);
+            return true;
+
+        } catch (ParseException e) {
+            return false;
+        }
+    }
+
+    Ad(String id, String description, String createdAt, String link, String vendor, String photoLink, List<String> hashTags, String discount,
```

Comment:
In a commercial project, a constructor with 10+ args won't be tolerated. Constructor is a method, and methods are expected to have 0, or 1, or 2, or 3 arguments at max. Want to pass more? consider implementing the very widespread builder pattern https://habr.com/ru/post/244521/, or use some external tool that hides initialization away (think about e.g. Lombok https://habr.com/ru/post/345520/.